### PR TITLE
docs(ct): api reference

### LINKED ISSUE #30581: [Docs]: Component testing API reference
### Page(s)

docs

### Description

It'd be great to have API docs for component testing. I could look into this if you could provide some pointers on where to place it?

- [x] add mount result docs: https://github.com/microsoft/playwright/issues/15919#issuecomment-2114583649

### PR DESCRIPTION
**Pull request by sand4rt**
closes: https://github.com/microsoft/playwright/issues/30581

### CODE REVIEW

File: docs/src/test-components-js.md [hunk 1]
```diff
@@ -195,93 +195,168 @@ npm run test-ct
 
 Refer to [Playwright config](./test-configuration.md) for configuring your project.
 
-## Test stories
+## API reference
```
Comment (dgozman) [inline docs/src/test-components-js.md:198]: I'd put this under "Stories" and "Under the hood" to make sure users learn the basics before diving into details of the API.

File: docs/src/test-components-js.md [hunk 2]
```diff
@@ -195,93 +195,168 @@ npm run test-ct
...
-    .setInputFiles('src/assets/logo.png');
+</TabItem>
 
-  await expect(component.getByAltText(/selected image/i)).toBeVisible();
-  await expect.poll(() => mediaSelected).toBe('logo.png');
+</Tabs>
+
+### children / slots
+
+Provide children/slots to a component when mounted.
```
Comment (dgozman) [inline docs/src/test-components-js.md:308]: Perhaps an example with a child component (as opposite to text child) would be nice?

**Review by dgozman — COMMENTED** [review]
Looks very good, just a few comments.

File: docs/src/test-components-js.md (hunk 2 above)
Comment (sand4rt) [reply docs/src/test-components-js.md:405]: Component as slot/child is not yet available across all frameworks: https://github.com/microsoft/playwright/issues/15900. Perhaps it's better to update this later on?

Comment (dgozman) [reply docs/src/test-components-js.md:405]: Yeah, makes sense.

**Review by dgozman — APPROVED** [review]

**Pull request merged**